Automattic/jetpack
# My Jetpack: Add support for Stats product

Fixes #31342 

## Proposed changes:

- Add Stats product support to My Jetpack
- Changed Security and Starter to use `Wpcom_Products::get_site_current_purchases` to reduce network requests

Next step:

- Replace the mocked WPCOM pricing with real pricing when ready

### Other information:


## Jetpack product discussion
n/a

## Does this pull request change what data or activity we track or use?
no

## Testing instructions:

* Build Jetpack `jetpack build plugins/jetpack`
* Open `/wp-admin/admin.php?page=my-jetpack`
* Open console in browser dev tools
* Ensure `myJetpackInitialState.products.items.stats` looks like the following

<img width="671" alt="image" src="https://github.com/Automattic/jetpack/assets/1425433/3accdfab-8d68-4487-8cd3-930c03fe3350">
